testutil/keeper/application.go

```diff
@@ -34,7 +34,9 @@ import (
 // WARNING: Using this map may cause issues if running multiple tests in parallel
 var stakedGatewayMap = make(map[string]struct{})
 
-func ApplicationKeeper(t testing.TB) (keeper.Keeper, context.Context) {
+// ApplicationKeeperWithSharedKeeper creates a new application keeper for testing
+// along with its dependencies returns the ApplicationKeeper, SharedKeeper and context.
+func ApplicationKeeperWithSharedKeeper(t testing.TB) (keeper.Keeper, types.SharedKeeper, context.Context) {
```

Comment:
This is not a good pattern and feels easy/lazy.

See the `TokenomicsModuleKeepers` pattern in `testutil/keeper/tokenomics.go` if we really need to embed and reuse different keepers.

Reply:
Fixed here dd5e7f19a719be447ca311e2fc209e36d42765e8